src/ucs/datastruct/conn_match.h

```diff
@@ -0,0 +1,217 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2020.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCS_CONN_MATCH_H_
+#define UCS_CONN_MATCH_H_
+
+#include <ucs/datastruct/khash.h>
+#include <ucs/datastruct/hlist.h>
+
+#include <inttypes.h>
+
+
+BEGIN_C_DECLS
+
+
+/**
+ * Connection sequence number
+ */
+typedef uint64_t ucs_conn_sn_t;
+
+
+/**
+ * Connection queue type
+ */
+typedef enum ucs_conn_match_queue_type {
+    /* Queue type for connections created by API and not
+     * connected to remote peer */
+    UCS_CONN_MATCH_QUEUE_EXP,
+    /* Queue type for connections created internally as
+     * connected to remote peer, but not provided to user yet */
+    UCS_CONN_MATCH_QUEUE_UNEXP,
+    /* Number of queues that are used by connection matching */
+    UCS_CONN_MATCH_QUEUE_LAST
+} ucs_conn_match_queue_type_t;
+
+
+/**
+ * Structure to embed in a connection entry to support matching with remote
+ * peer's connections
+ */
+typedef struct ucs_conn_match_elem {
+    ucs_hlist_link_t          list;          /* List entry into endpoint
+                                                matching structure */
+} ucs_conn_match_elem_t;
+
+
+/**
+ * Connection peer entry - allows *ordered* matching of connections between
+ * a pair of connected peers. The expected/unexpected lists are *not* circular
+ */
+typedef struct ucs_conn_match_peer {
```

Comment:
maybe move this struct to C file? since it's "private"

Reply:
done